src/main/java/build/buildfarm/instance/shard/Operations.java

```diff
@@ -154,7 +154,7 @@ public void insert(UnifiedJedis jedis, String invocationId, String name, String
 
     // We also store a mapping from invocationID -> operationIDs
     // This is a common lookup that needs to be performant.
-    if (invocationId != "" && jedis.sadd(invocationId, name) == 1) {
+    if (!"".equals(invocationId) && jedis.sadd(invocationId, name) == 1) {
```

Comment:
That assumes `invocationId != null` :)
```suggestion
    if (!invocationId.isEmpty() && jedis.sadd(invocationId, name) == 1) {
```